Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -314,15 +311,34 @@ void MainWindow::Pause()
   emit EmulationPaused();
 }
 
-bool MainWindow::Stop()
+void MainWindow::OnStopComplete()
+{
+  m_stop_requested = false;
+  m_render_widget->hide();
+
+  if (m_exit_requested)
+    QGuiApplication::instance()->quit();
+
+  // If the current emulation prevented the booting of another, do that now
+  if (!m_pending_boot.isEmpty())
+  {
+    StartGame(m_pending_boot);
+    m_pending_boot.clear();
+  }
+}
+
+bool MainWindow::RequestStop()
 {
   if (!Core::IsRunning())
+  {
+    Core::QueueHostJob([this] { OnStopComplete(); }, true);
```

Comment:
This is needed so that `OnStopComplete` is always called **after** `RequestStop` has returned, which is nice and consistent. A more concrete reason appears in `eventFilter`:

```c++
if (RequestStop() && object == this)
  m_exit_requested = true;
```

wouldn't work when a game isn't running.

(This is a pretty common pattern in NodeJS, if you're into that, where code uses `process.nextTick(func)` to make sure that the sequencing of events is consistent and easy to reason about.)